mne/_fiff/reference.py

```diff
@@ -155,6 +205,31 @@ def _apply_reference(inst, ref_from, ref_to=None, forward=None, ch_type="auto"):
     return inst, ref_data
 
 
+def _apply_dict_reference(inst, ref_dict):
+    """Apply a dict-based custom EEG referencing scheme."""
+    _check_before_dict_reference(inst, ref_dict)
+
+    # Copy the data instance to re-reference:
+    ref_to_data = inst.copy()._data
+    if len(ref_dict) > 0:
+        # Loop through each channel to re-reference:
+        for ch in ref_dict.keys():
+            assert len(ref_dict[ch]) > 0, f"No channel to re-reference ch-{ch}"
+            # Get indices of the channels to use as reference
+            ref_from = pick_channels(inst.ch_names, ref_dict[ch], ordered=True)
+            # Get indice of channel to re.reference:
+            ref_to = pick_channels(inst.ch_names, ch, ordered=True)
```

Comment:
Can add a warning around here if ref_from and ref_to belong to different channel types